pkg/imgpkg/bundle/bundle_images_lock.go

```diff
@@ -32,28 +34,38 @@ func (o *Bundle) buildAllImagesLock(processedImgs map[string]struct{}) (*ImagesL
 
 	allImagesLock := NewImagesLock(imagesLock, o.imgRetriever, o.Repo())
 
+	errChan := make(chan error, len(imagesLock.Images))
+	throttle := util.NewThrottle(concurrency)
```

Comment:
Yes, I thought about it, but the problem is that if you select the concurrency to 1 and have nested bundles the worker from the outer bundle will be blocked until the inner ones are resolved. Given that we only have 1 worker this will be blocked forever.
So I decided to incur this penalty and ignore the amount of concurrency you asked for in the command line.  Basically, you can specify how many concurrent workers you want running for all Images in a particular Bundle from the command line. So if you have 5 levels of nested bundles you will get 5 x ConcurrencyValueProvided

Reply:
good point about blocking. i still think that concurrency should be against total requests in progress. feels problematic if the input (nested bundles) can affect amount of stuff being downloaded at once.

can we not throttle.Take/Done around the actual fetching and not just against the entire go routine?

Reply:
I pushed that down near the request now we should be able to throttle the request only